### PR TITLE
fix: support deployment summaries, speeding up deployments page

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Makes use of https://github.com/ibi-group/datatools-server/pull/486. Speeds up the deployment page by not loading in entire deployment object for every entry. Also makes some updates to the mocked server to add a deployment summary endpoint.

Blocked by https://github.com/ibi-group/datatools-server/pull/491

### CODE REVIEW

File: __tests__/test-utils/mock-data/manager.js [hunk 1]
```diff
@@ -13,6 +13,23 @@ import type {
 
 let COUNTER = 0
 
+/**
+ * Make a mock deployment given a project and some FeedVersions. This is a
```
Comment (philip-cline) [inline __tests__/test-utils/mock-data/manager.js]: I suppose you could say 
```suggestion
 * Make a mock deployment summary given a project and some FeedVersions. This is a
```

File: lib/manager/components/deployment/DeploymentViewer.js [hunk 2]
```diff
@@ -76,6 +76,10 @@ export default class DeploymentViewer extends Component<Props, State> {
   }
 
   componentDidMount () {
+    const { deployment, fetchDeployment } = this.props
+    // If there is no project id, we are dealing with a summary and need to fetch the
```
Comment (philip-cline) [inline lib/manager/components/deployment/DeploymentViewer.js]: Should there be a check here to see if there is no project id?

**Review by philip-cline — APPROVED** [review]
Nice solution to the project ID issue. Just a few small questions, looks good to me

File: __tests__/test-utils/mock-data/manager.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply __tests__/test-utils/mock-data/manager.js]: Good catch thank you! Adjusted.

File: lib/manager/components/deployment/DeploymentViewer.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply lib/manager/components/deployment/DeploymentViewer.js]: Yep thanks for catching this! This comment was outdated.

File: __tests__/test-utils/mock-data/manager.js [hunk 3]
```diff
@@ -13,6 +13,23 @@ import type {
 
 let COUNTER = 0
 
+/**
+ * Make a mock deployment summary given a project and some FeedVersions. This is a
+ * function so that circular references can be defined.
```
Comment (binh-dam-ibigroup) [inline __tests__/test-utils/mock-data/manager.js:18]: Is there a reason the function args are not used?

File: lib/manager/reducers/projects.js [hunk 4]
```diff
@@ -346,7 +346,10 @@ const projects = (state: ProjectsState = defaultState, action: Action): Projects
       })
     }
     case 'RECEIVE_DEPLOYMENTS': {
-      const {deployments, feedSourceId, projectId} = action.payload
+      const {deployments: deploymentsSummaries, feedSourceId, projectId} = action.payload
+      // Deployment summaries don't include a project id, so we need to add one
+      const deployments = deploymentsSummaries.map(d => ({...d, projectId}))
```
Comment (binh-dam-ibigroup) [inline lib/manager/reducers/projects.js]: The new `deploymentSummaries` comes with a new field `numberOfFeedVersions` that should now go under the "# of feeds" column in the `DeploymentListItem` component. As written in the PR, the "# of feeds" column in the list of deployments shows N/A until the full deployment object is fetched.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Looks neat, but the "# of feeds" column needs to be populated from the new `DeploymentSummarY` object.

File: __tests__/test-utils/mock-data/manager.js (hunk 3 above)
Comment (miles-grant-ibigroup) [reply __tests__/test-utils/mock-data/manager.js:18]: Good catch thanks! Parameters have been removed

File: lib/manager/reducers/projects.js (hunk 4 above)
Comment (miles-grant-ibigroup) [reply lib/manager/reducers/projects.js]: Thanks for catching this one! Appreciate the heads up about `numberOfFeedVersions`. We now create a fake array.

File: lib/manager/reducers/projects.js [hunk 5]
```diff
@@ -346,7 +346,11 @@ const projects = (state: ProjectsState = defaultState, action: Action): Projects
       })
     }
     case 'RECEIVE_DEPLOYMENTS': {
-      const {deployments, feedSourceId, projectId} = action.payload
+      const {deployments: deploymentsSummaries, feedSourceId, projectId} = action.payload
+      // Deployment summaries don't include a project id, so we need to add one
+      // Deployment summaries don't include feedversions array, so we create a fake array that gets replaced later
+      const deployments = deploymentsSummaries.map(d => ({...d, projectId, feedVersions: Array(d.numberOfFeedVersions)}))
```
Comment (binh-dam-ibigroup) [inline lib/manager/reducers/projects.js:352]: Using an array of `undefined` causes a crash when the user navigates from the Deployments tab back to the Feed Sources tab using the browser's Back button (clicking the Feed Sources tab still works).

The error message is as follows:
```
Uncaught TypeError: Cannot read properties of undefined (reading 'feedSource')
    at version.js:176:28
    at Array.find (<anonymous>)
    at getVersionValidationSummaryByFilterStrategy (version.js:175:67)
    ...
```

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Unfortunately the latest change results in a crash when navigating back to the Feed Sources tab, so I'm requesting that to be fixed.

File: lib/manager/reducers/projects.js (hunk 5 above)
Comment (miles-grant-ibigroup) [reply lib/manager/reducers/projects.js:352]: I've fixed this bug, but there is a problem! Since the new "compressed" data doesn't include feed source ids, it's not possible to populate the deployed version column! 

Is there something I'm missing?

Comment (binh-dam-ibigroup) [reply lib/manager/reducers/projects.js:352]: Probably an omission on the server side?

Comment (miles-grant-ibigroup) [reply lib/manager/reducers/projects.js:352]: Ok so then not something I'm missing then. Do we want to include that though? The list of feeds in each deployment can get quite large especially if there are a lot of deployments with lots of feeds.

I'll schedule a time for us to figure this out offline.

**Comment by miles-grant-ibigroup** [comment]
Have just run the e2e tests locally and they are failing, suggesting we need to fix the feed source columns

**Comment by miles-grant-ibigroup** [comment]
Blocked pending feedSource endpoint updates on the backend.

**Comment by miles-grant-ibigroup** [comment]
Updated to make use of new fields in https://github.com/ibi-group/datatools-server/pull/491. Will wait for e2e tests to pass and then assign to @binh-dam-ibigroup. 

Even though the startDate field is missing, I have defaults set up so it shouldn't impact behavior (aside from the one edge case where the status label will fail to say that a feed which starts in the future starts in the future).

File: lib/manager/components/version/FeedVersionTabs.js [hunk 6]
```diff
@@ -107,8 +106,8 @@ export default class FeedVersionTabs extends Component<Props> {
             <Row>
               {countFields.map(c => {
                 const summary = version.validationSummary
-                const diff = comparedVersion ? (summary[c] - comparedVersion.validationSummary[c]) : 0
-                const value = summary[c]
+                const diff = comparedVersion ? ((summary[c] || 0) - (comparedVersion.validationSummary[c] || 0)) : 0
+                const value = summary[c] || 0
```
Comment (binh-dam-ibigroup) [inline lib/manager/components/version/FeedVersionTabs.js]: Maybe swap these two rows and reuse variable `value`.

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

File: lib/manager/components/version/FeedVersionTabs.js (hunk 6 above)
Comment (miles-grant-ibigroup) [reply lib/manager/components/version/FeedVersionTabs.js]: Good catch! Refactored in 15a6f1ad6aa68131ffe4fc4b026bef670d2b1eac
